src/test/ui/type-check/issue-17283.rs

```diff
@@ -24,25 +24,25 @@ fn main() {
     // `x { ... }` should not be interpreted as a struct literal here
```

Comment:
This isn't really a 'regression test for obscure scenario'. Let's rename the test to `assignment-in-if.rs`. While we're at it, can you add some other obscure cases, like this one:

```rust
if (if true { x = 4 } else { x = 5 }) { ... }
```

I don't really care what message it emits, although I *think* (if I read the code right...) that the expectation will be propagated. I just wanted to test it.